File: app/src/main/java/io/github/benoitduffez/cupsprint/ssl/AndroidCupsHostnameVerifier.java
package io.github.benoitduffez.cupsprint.ssl;

import javax.net.ssl.HostnameVerifier;
import javax.net.ssl.SSLSession;

import io.github.benoitduffez.cupsprint.CupsPrintApp;
import io.github.benoitduffez.cupsprint.app.HostNotVerifiedActivity;

/**
 * Used with {@link HostNotVerifiedActivity} to trust certain hosts
 */
public class AndroidCupsHostnameVerifier implements HostnameVerifier {
    @Override
    public boolean verify(String hostname, SSLSession session) {
        return HostNotVerifiedActivity.isHostnameTrusted(CupsPrintApp.getContext(), hostname);
    }
}

File: app/src/main/java/io/github/benoitduffez/cupsprint/ssl/AndroidCupsHostnameVerifier.kt
package io.github.benoitduffez.cupsprint.ssl

import android.content.Context
import io.github.benoitduffez.cupsprint.app.HostNotVerifiedActivity
import javax.net.ssl.HostnameVerifier
import javax.net.ssl.SSLSession

/**
 * Used with [HostNotVerifiedActivity] to trust certain hosts
 */
class AndroidCupsHostnameVerifier(val context: Context) : HostnameVerifier {
    override fun verify(hostname: String, session: SSLSession): Boolean =
            HostNotVerifiedActivity.isHostnameTrusted(context, hostname)
}
